mlrun/runtimes/utils.py

```diff
@@ -315,6 +315,7 @@ def apply_kfp(modify, cop, runtime):
         cop.volumes.clear()
         cop.container.volume_mounts.clear()
 
+    runtime.spec.mount_applied = True
```

Comment:
The idea of the apply is to allow users to create modifiers of their own, I do not think someone will ever use that, but still, not every modifier is a mount, should be easy to check the modifier and only if it's actually a mount modifier do the above

Reply:
I know, I thought about that caveat. However, finding out if a modifier is a "mount" is not that simple. Note that some of the modifiers don't even do a real mount - `v3io_cred` just sticks env variables to the spec and that's it. However, if someone already used `func.apply(v3io_cred())` I would not want to auto-mount. 
So, based on the fact that it's complicated to figure out if a mount was applied or not, I decided for now to simply mark the runtime like that whenever an `apply()` is called.

Reply:
Why it's not simple, can't you just check whether the modifier is one of the known mount modifier functions ?

Reply:
oh, that's what you mean - I can do that, but then if someone adds another mount modifier this list will need to be updated somehow, and I'm pretty sure nobody will remember this needs to be done.